fix(router): add condition on href rewrite if routerLink has null value

maximelafarie opened the pull request:
Fix a bug where a null value passed to routerLink directive overrides the existing href.

fixes #26040


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature


## What is the current behavior?
Description and full Stackblitz example are in the issue below: 

Issue Number: #26040 


## What is the new behavior?
If a `null` value is passed to `routerLink` directive and if an `href` attribute is defined and not empty, `href` will remain unchanged. 


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

Commit:
fix(router): add condition on href rewrite if routerLink has null value

Fix a bug where a null value passed to routerLink directive overrides the existing href.

fixes #26040

Commit:
fix(router):  tests and indentation

Commit:
fix(router): remove non pertinent stuff and re indent with gulp format

Commit:
fix(router): Fix bad tests and simplify condition

maximelafarie commented:
I ran the test suite as described [here](https://github.com/angular/angular/blob/master/docs/DEVELOPER.md#running-tests-locally) and [here](https://github.com/angular/angular/blob/master/docs/DEBUG.md) but it doesn't crash any error as the CI does, so I can't find how to get the same result as the above CIs. It would be awesome if someone can explain me how to do that. 😖

ocombe commented:
Those documents haven't been updated recently, we need to add the commands for ivy (cc @IgorMinar).
To run the tests for ivy you need to use the following command: `yarn test-ivy-aot //packages/router/...`. It requires that you install bazel on your machine.

manekinekko commented:
These tests are also failing on my machine while running the `./test.sh router`. So the CI seems to be fine.

Commit:
fix(router): Get the attr.href value and clean tests

Commit:
fix(router): little rollback and fix case of null url in router state

maximelafarie commented:
Thank you very much @ocombe and @manekinekko, tests are well now! 😄

maximelafarie commented:
@jasonaden could you take a look at this PR please? I rebased it and fixed conflicts but it would be awesome to review it before this branch becomes obsolete. At your disposal! 😄

maximelafarie commented:
All is ok and up to date! ✅

But the CI can't capture browsers with 
```
yarn karma start ./karma-js.conf.js --single-run --browsers=${KARMA_JS_BROWSERS}
```
I guess it's a CI-related error. It would be my pleasure to rerun the CI's workflow, but I'm not allowed to do it. 😉

maximelafarie commented:
Hi @jasonaden ! ✋ What's the progress on this PR?

LouisAugry commented:
You should squash commits ! 😀Or simply recreates one PR, this one seems to be lost in in the depths.

Maybe @IgorMinar @manekinekko @ocombe can do the review or help in any way?
